app/javascript/components/ssr-rsc-playground/ui/SsrWalkthrough.tsx

```
@@ -0,0 +1,764 @@
+      setEffMs(startEff + effDelta * easeInOutSine(progress));
+
+      if (progress < 1) {
+        rafRef.current = requestAnimationFrame(tick);
+      } else {
+        setEffMs(targetEff);
+        setAnimating(false);
+        setTimeout(() => setNoteVisible(true), 200);
+      }
+    };
+    rafRef.current = requestAnimationFrame(tick);
+  }, [stepIndex, effMs, animating, stopRaf]);
```

ESLint

[error] React Hook useCallback has an unnecessary dependency: 'stopRaf'. Either exclude it or remove the dependency array.

(react-hooks/exhaustive-deps)